linux/devdoc/sync_linux_requirements.md

```diff
@@ -0,0 +1,45 @@
+# sync linux
+================
+
+## Overview
+
+`sync linux` is the Linux implementation of the `sync` header using [futex](https://www.man7.org/linux/man-pages/man2/futex.2.html).
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout);
+MOCKABLE_FUNCTION(, void, wake_by_address_all, void*, address);
+MOCKABLE_FUNCTION(, void, wake_by_address_single, void*, address);
+```
+
+## wait_on_address
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout)
+```
+
+**SRS_SYNC_LINUX_43_001: [** `wait_on_address` shall define a `timespec` struct with `.tv_nsec` equal to `timeout* 10^6`. **]**
+
+
+**SRS_SYNC_LINUX_43_002: [** `wait_on_address` shall call `syscall` from `sys/syscall.h` with arguments `SYS_futex`, `address`, `FUTEX_WAIT`, `*compare_address`, `timeout_struct`, `NULL`, `NULL`. **]**

```

Comment:
>SYS_futex [](start = 105, length = 9)

We should use `FUTEX_PRIVATE_FLAG` since we don't do this cross-process.

Reply:
(and can't do it across process for Windows)

---
In reply to: [441218548](https://github.com/Azure/azure-c-pal/pull/5#discussion_r441218548) [](ancestors = 441218548)